internal/sops/gcpkms/keysource.go

```diff
@@ -0,0 +1,163 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package gcpkms
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"regexp"
+	"time"
+
+	cloudkms "google.golang.org/api/cloudkms/v1"
```

Comment:
Just discovered this package is marked as deprecated: https://pkg.go.dev/google.golang.org/api/cloudkms/v1. Instead, we should use https://pkg.go.dev/cloud.google.com/go/kms/apiv1. This is inline with our Azure KMS solution which also uses a different (newer) SDK than SOPS itself.

Reply:
I used that at first, but the secrets encrypted by sops weren't being decrypted.

Reply:
Maybe the decrypt and/or encrypt request default settings changed between packages, the decryption notice itself seems to suggest it's backwards compatible (as well as the API version of the service).

Reply:
I will look into it again to see if I can get it to work.